pkg/controller/vsphere/machine_scope.go

```diff
@@ -162,12 +163,58 @@ func (s *machineScope) checkNodeReachable() (bool, error) {
 		}
 		return false, err
 	}
+	return nodeReachable(node), nil
+}
+
+// deleteUnevictedPods checks respective node for reachability,
+// if the node is not reachable it tries to remove pods in the 'Terminating' state.
+// Returns the number of deleted pods and errors if there were any.
+// Returns error, if the node is operational.
+func (s *machineScope) deleteUnevictedPods() (int, error) {
+	node, err := s.getNode()
+	if err != nil {
+		// do not return error if node object not found, treat it as unreachable
+		if apimachineryerrors.IsNotFound(err) {
+			return 0, nil
+		}
+		return 0, err
+	}
+	if nodeReachable(node) {
+		return 0, fmt.Errorf("node is in operational state, won't proceed with pods deletion")
+	}
+
+	terminatingPods, err := getPodList(s.Context, s.apiReader, node, []podPredicate{isTerminating})
+	if err != nil {
+		return 0, err
+	}
+
+	gracePeriodSeconds := int64(0)
+	deleteOptions := &runtimeclient.DeleteOptions{
+		GracePeriodSeconds: &gracePeriodSeconds,
+	}
+	deletedPods := 0
```

Comment:
Does this mean they get removed straight away? Even in a terminating state? Does this skip kubelet acknowledging that the pod has been removed?

Reply:
Yes, this removes pods straight away. It is basically `--force` kubectl's flag analog.

https://github.com/kubernetes/kubernetes/blob/75ff830ab198348b0c727d05e54754d5f73a7286/staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go#L191

Reply:
I would like to leave it with a delay for now and start backporting this. Then we could look into that taint, and introduce it in followup PR.